qiita_db/analysis.py

```diff
@@ -958,7 +983,78 @@ def _build_biom_tables(self, grouped_samples, rename_dup_samples=False):
                     new_table.to_hdf5(
                         f, "Generated by Qiita, analysis id: %d, info: %s" % (
                             self._id, label))
-                biom_files.append((data_type, biom_fp))
+
+                # post_processing_cmds are a list of commands to run on the
+                # final BIOM. The order of operations is list-order. Each
+                # element of the list is a dictionary containing the Conda env
+                # to use, the script to run, and a dictionary of parameters.
+                if post_processing_cmds:
+                    # assuming all commands require archives, obtain
+                    # archives once, instead of for every cmd.
+                    features = load_table(biom_fp).ids(axis='observation')
+                    features = list(features)
+                    archives = Archive.retrieve_feature_values(
+                        archive_merging_scheme=archive_merging_scheme,
+                        features=features)
+
+                    # remove archives that SEPP could not match
+                    archives = {f: loads(archives[f])
+                                 for f, plc
+                                 in archives.items()
+                                 if plc != ''}
+
+                    # since biom_fp uses base_fp as its location, assume it's
+                    # suitable for other files as well.
+                    output_dir = base_fp
+
+                    fp_archive = '%s/archive.json' % output_dir
```

Comment:
sorry for not catching this before: you should use join to create this path and also use the analysis id to be safe something like: `fp_archive = join(output_dir, 'archive_%d.json' % self.id)`

Reply:
Ah gotcha, platform independence path creation.